src/common/task.cc

```diff
@@ -359,45 +456,60 @@ bool TaskSpec_is_dependent_on(TaskSpec *spec, ObjectID object_id) {
       }
     }
   }
+  // Iterate through the execution dependencies to see if it contains object_id.
+  for (auto dependency_id : execution_dependencies_) {
```

Comment:
this makes a copy of each ObjectID element of the execution dependencies vector. Why not 
``` 
for (const auto &dep_id : exe_deps) {
```
ObjectID_equal is a readonly consumer (which, incidentally, we should be enforcing in the function prototype -- I can create a PR for that separately).

Reply:
a side note (less important for now), if we ever envision large numbers of execution dependencies, we should probably make it a hashmap. We had to eliminate linear searches in the past for performance reasons.